src/Type/PHPUnit/CreateMockDynamicReturnTypeExtension.php

```diff
@@ -39,17 +39,19 @@ public function getTypeFromMethodCall(MethodReflection $methodReflection, Method
 			return $methodReflection->getReturnType();
 		}
 		$arg = $methodCall->args[$argumentIndex]->value;
-		if (!($arg instanceof \PhpParser\Node\Expr\ClassConstFetch)) {
-			return $methodReflection->getReturnType();
-		}
+		if ($arg instanceof \PhpParser\Node\Expr\ClassConstFetch) {
+			$class = $arg->class;
+			if (!($class instanceof \PhpParser\Node\Name)) {
+				return $methodReflection->getReturnType();
+			}
 
-		$class = $arg->class;
-		if (!($class instanceof \PhpParser\Node\Name)) {
+			$class = (string) $class;
+		} elseif ($arg instanceof \PhpParser\Node\Scalar\String_) {
```

Comment:
Please update this with the latest advancements in PHPStan's typesystem. See https://github.com/phpstan/phpstan/commit/a46cbc067a6df0f8d36732ccd2f11b643124da4d for an example.

Please also move the fixes of changes in PHPStan's dev-master to a separate commit.

Reply:
done